src/basic/path-util.c

```diff
@@ -690,6 +690,9 @@ int find_executable_full(const char *name, const char *root, bool use_path_envva
         if (!p)
                 p = DEFAULT_PATH;
 
+        if (exec_search_paths)
+                p = strv_join(exec_search_paths, ":");
```

Comment:
oom check missing

also, the block immediately above seems unecessary if you override it here anyway, hence swap this and make it conditional via an else block

Reply:
I moved it above. The code no longer uses strv_join though, so not sure oom check is necessary here anymore. The new code will just go through the strv and check each of those paths rather than joining into another string.